src/utils/fits.rs

```diff
@@ -85,8 +88,8 @@ pub fn buffer_to_image(buffer: &[u8]) -> Result<Vec<f32>, Box<dyn std::error::Er
     // so if NAXIS1 is not NAXIS_STANDARD, we need to add NAXIS_STANDARD - NAXIS1 zeros to the start and end of each row
     // and if NAXIS2 is not NAXIS_STANDARD, we need to add NAXIS_STANDARD - NAXIS2 zeros to the start and end of the vector
 
-    let offset1 = (NAXIS_STANDARD - naxis1) / 2; // Assuming naxis1 and naxis2 are both usize
-    let offset2 = (NAXIS_STANDARD - naxis2) / 2;
+    let offset1 = ((NAXIS_STANDARD - naxis1) as f32 / 2.0).ceil() as usize;
```

Comment:
I had to change the code here, because what we had would behave incorrectly in some cases when the NAXIS wasn't an even number, or just the max size - 1. That deals with those errors by first conversion to f32 before doing the division, and then using ceil.